mewarpx/mewarpx/assemblies.py

```diff
@@ -371,8 +387,114 @@ def calculatenormal(self, px, py, pz):
+        self.center_y = 0.0
+
+        # sanity check
+        if self.r_outer == 0:
+            raise AttributeError('Cannot have a cylinder with 0 outer radius.')
+
+        # check if the cylinder forms the simulation boundary
+        if self.r_inner == mwxrun.rmax:
+            # set solver boundary potential
+            mwxrun.grid.potential_xmax = self.V
+            # set label to correctly grab scraped particles from the buffer
+            self.scraper_label = 'x_hi'
+        elif self.r_outer == mwxrun.rmin:
+            # set solver boundary potential
+            mwxrun.grid.potential_xmin = self.V
+            # set label to correctly grab scraped particles from the buffer
+            self.scraper_label = 'x_lo'
+        else:
+            # negative means outside of object
+            self.implicit_function = (
+                f"-(abs(x-{self.r_center})-{self.r_outer - self.r_center})"
+            )
+            # set label to correctly grab scraped particles from the buffer
+            self.scraper_label = 'eb'
+            # install the EB in the simulation
+            self._install_in_simulation()
+
+    def isinside(self, X, Y, Z, aura=0):
+        """
+        Calculates which grid tiles are within the cylinder.
+
+        Arguments:
+            X (np.ndarray): array of x coordinates of flattened grid.
+            Y (np.ndarray): array of y coordinates of flattened grid.
+            Z (np.ndarray): array of z coordinates of flattened grid.
+            aura (float): extra space around the conductor that is considered
+                inside. Useful for small, thin conductors that don't overlap any
+                grid points. In units of meters.
+
+        Returns:
+            result (np.ndarray): array of flattened grid where all tiles
+                inside the cylinder are 1, and other tiles are 0.
+        """
+        dist_to_center = np.sqrt(
+            (X - self.center_x)**2 + (Y - self.center_y)**2
+        )
+        dr = self.r_outer + aura - self.r_center
+        result = np.where(abs(dist_to_center - self.r_center) <= dr, 1, 0)
+
+        return result
+
+    def calculatenormal(self, px, py, pz):
+        """
+        Calculates Normal of particle inside/outside of conductor to nearest
+        surface of conductor
+
+        Arguments:
+            px (np.ndarray): x-coordinate of particle in meters.
+            py (np.ndarray): y-coordinate of particle in meters.
+            pz (np.ndarray): z-coordinate of particle in meters.
+
+        Returns:
+            nhat (np.ndarray): Normal of particles of conductor to nearest
+                surface of conductor.
+        """
+        # distance from center of cylinder
+        dist = np.sqrt((px - self.center_x)**2 + (py - self.center_y)**2)
+        nhat = np.zeros([3, len(px)])
+        nhat[0, :] = (px - self.center_x) / dist
+        nhat[2, :] = (pz - self.center_z) / dist
+
+        # flip the normal vector for points inside the cylinder
```

Comment:
```suggestion
        # Flip the normal vector for points inside the cylinder inner wall,
        # or inside the cylinder assembly but closer to the outer wall than the inner.
```
it looks like is what the below statement is doing?